build/gen_stub.php

```diff
@@ -39,13 +49,13 @@ function processStubFile(string $stubFile, Context $context) {
         $oldStubHash = extractStubHash($arginfoFile);
         if ($stubHash === $oldStubHash && $context->forceRegeneration === false) {
             /* Stub file did not change, do not regenerate. */
-            return;
+            return null;
         }
 
         initPhpParser();
         $fileInfo = parseStubFile($stubCode);
         $arginfoCode = generateArgInfoCode($fileInfo, $stubHash);
-        if (file_put_contents($arginfoFile, $arginfoCode)) {
+        if ($context->forceRegeneration === true && file_put_contents($arginfoFile, $arginfoCode)) {
```

Comment:
I'm using `$context->forceRegeneration === true` to try to emphasize that it has 3 states now

Reply:
Not a big fan of bool|null ^^ Maybe split this up into two flags, or use an enum?

Reply:
Yeah, that's ok for me!

Reply:
It looks much nicer now! Thanks for the comment.